apps/gitness/src/pages/project-settings/project-settings-general-page.tsx

```diff
@@ -93,7 +98,9 @@ export const ProjectSettingsGeneralPage = () => {
     deleteSpaceMutation.mutate(
       { space_ref: space?.path },
       {
-        onSettled: () => setIsDeleting(false) // Ensure isDeleting is reset after the mutation completes
+        onSettled: () => {
+          setDeleteError(null)
+        } // Ensure isDeleting is reset after the mutation completes
```

Comment:
Shouldn't we reset `onSuccess` ?

Reply:
It is not necessary to reset onSuccess inside onSettled. Both onSuccess and onSettled are lifecycle methods for handling the state of a mutation, but they serve different purposes.

Key Differences Between onSuccess and onSettled:
onSuccess: This is called only when the mutation succeeds. It is where you typically handle any logic that should occur after a successful API call, such as updating local state or redirecting the user.

onSettled: This is called regardless of whether the mutation succeeds or fails. It’s useful for doing any cleanup tasks that should occur after the mutation completes, such as resetting UI elements, loading states, or clearing error messages.

Reply:
Wouldn't it be better to set the error to null only when the mutation succeeds, rather than resetting it regardless of the outcome, as is done with onSettled?

Reply:
Yeah I can set up this line in the `deleteMutation onSuccess`

Reply:
Fixed